fix: move table create update delete to query route

sweatybridge opened the pull request:
## What kind of change does this PR introduce?

refactor

## What is the current behavior?

- mirrors implementation of [pgmeta](https://github.com/supabase/postgres-meta/blob/master/src/lib/PostgresMetaTables.ts#L122) exactly
- adjusts update query to fetch primary keys implicitly

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

sweatybridge committed:
fix: move table create update delete to query route

avallete commented on packages/pg-meta/src/pg-meta-tables.ts:
```diff
@@ -196,103 +178,76 @@ function update(
+  SELECT conname
+    INTO r
+    FROM pg_constraint
+    WHERE contype = 'p' AND conrelid = ${literal(old.id)};
```
**question**

If we wanna get rid of the `id` dependencies for migrations tracking should this be replaced with a select subquery to get the table id from `name+schema` ?

sweatybridge replied:
Currently the UI relies on table id in path param for navigation. It will be quite complicated to move away from it.

alaister reviewed (commented):
I get the following error when trying to add a column in the table edit ui:
<img width="383" alt="Screenshot 2025-05-14 at 15 55 58" src="https://github.com/user-attachments/assets/f3e5cb07-222d-41e4-b199-492db9d24aec" />

I also got an error the first time I tried creating a table, but I can't seem to replicate that again :/

sweatybridge committed:
chore: implement query to fetch a single table

saltcod commented:
renaming a table gives me this:
![screenshot-2025-05-14-at-11 04 48](https://github.com/user-attachments/assets/3c27e584-c1d7-4cbb-8e88-5dcfbe1e47f3)

saltcod commented:
![screenshot-2025-05-14-at-11 06 32](https://github.com/user-attachments/assets/82b8dd27-17ca-406f-90d4-529c3fca26ad)

saltcod commented:
disabling RLS fails with the same error as the rest 

![screenshot-2025-05-14-at-11 07 21](https://github.com/user-attachments/assets/c5e8e7e8-7c47-408c-affe-d509d3c496ba)

avallete committed:
fix: retrieve table after update

sweatybridge commented on apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/SidePanelEditor.utils.tsx:
```diff
@@ -670,14 +685,28 @@ export const updateTable = async ({
+    ),
+    queryFn: ({ signal }) =>
+      getTable(
+        { projectRef, connectionString, name: payload.name, schema: payload.schema },
```
This needs a ternary as well

sweatybridge committed:
chore: assign type to update table payload

sweatybridge committed:
Merge branch 'master' into qiao/table

sweatybridge committed:
chore: use updated table columns for edit

avallete committed:
chore: make executeSql castable with generic (#35685)

avallete committed:
Chore/refactor derivate more types from queries (#35687)

* chore: make executeSql castable with generic

* chore: derivate types from performed queries

- It allows to decouple more the frontend logic and the pg-meta/sql-query logic allowing to reduce the number of cast
and get closer types between what we do fetch and what we expect in our components

sweatybridge committed:
Merge branch 'master' into qiao/table

avallete committed:
fix: remove existing check

avallete committed:
Merge branch 'master' into qiao/table

avallete committed:
Merge branch 'master' into qiao/table

avallete committed:
Merge branch 'master' into qiao/table

avallete committed:
Merge branch 'master' into qiao/table

sweatybridge commented on apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnEditor.utils.ts:
```diff
@@ -126,7 +127,7 @@ export const generateUpdateColumnPayload = (
   const comment = ((field.comment?.length ?? '') === 0 ? '' : field.comment)?.trim()
 
-  const check = field.check?.trim()
+  const check = field.check?.trim() ? field.check?.trim() : undefined
```
what is this supposed to do?

sweatybridge replied:
Based on https://github.com/supabase/supabase/blob/master/packages/pg-meta/src/pg-meta-columns.ts#L341, we should set empty string to null.

```
const check = field.check?.trim() || null
```

sweatybridge commented on apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/ColumnEditor/ColumnEditor.utils.ts:
```diff
@@ -126,7 +127,7 @@ export const generateUpdateColumnPayload = (
   const type = field.isArray ? `${field.format}[]` : field.format
   const comment = ((field.comment?.length ?? '') === 0 ? '' : field.comment)?.trim()
```
```suggestion
  const comment = field.comment?.trim() || null
```

ditto

sweatybridge committed:
chore: handle null comment and check

sweatybridge committed:
fix: format check name as identifier

saltcod commented:
Tested every table action I could think of and looks good on staging!
